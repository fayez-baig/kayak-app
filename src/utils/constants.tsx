export const ALLIANCES = [
  {
    code: "OW",
    name: "Oneworld",
  },
  {
    code: "ST",
    name: "Sky Team",
  },
  {
    code: "SA",
    name: "Star Alliance",
  },
];

export const ITEMS_PER_PAGE = 12;

export const INITIAL_PAGE_NUMBER = 1;

export const FILTER_TITLE = "Oneworld";

export const LOADER_TEST_ID = "suspense-loader";

export const ERROR_MESSAGE = "Something went wrong !!";

export const NEXT_BUTTON_TEST_ID = "next-btn";

export const PREV_BUTTON_TEST_ID = "prev-btn";

export const ERROR_BOUNDARY_MESSAGE = "Oops !";
