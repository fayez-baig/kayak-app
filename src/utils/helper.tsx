export const alliances = [
  {
    code: "OW",
    name: "Oneworld",
  },
  {
    code: "ST",
    name: "Sky Team",
  },
  {
    code: "SA",
    name: "Star Alliance",
  },
];

export const getFilteredData = (data, checkedValues) => {
  return !checkedValues.length
    ? data?.data
    : data?.data?.filter(({ alliance }) => checkedValues.includes(alliance));
};
