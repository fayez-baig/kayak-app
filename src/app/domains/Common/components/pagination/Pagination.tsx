import { FC, ReactElement } from "react";
import { PaginationProps } from "./types";
import { NEXT_BUTTON_TEST_ID, PREV_BUTTON_TEST_ID } from "utils/constants";
import "./pagination.scss";

const Pagination: FC<PaginationProps> = ({
  itemsPerPage,
  totalAirlinesData,
  paginate,
  currentPage,
}): ReactElement => {
  const pageNumbers: number[] = [];

  for (let i = 1; i <= Math.ceil(totalAirlinesData / itemsPerPage); i++) {
    pageNumbers.push(i);
  }

  const totalPage = pageNumbers.length - 1;

  return (
    <div className="pagination-container">
      <button
        type="button"
        data-testid={PREV_BUTTON_TEST_ID}
        className="btn"
        onClick={() => paginate(currentPage - 1)}
        disabled={currentPage === 1}
      >
        &lt;
      </button>
      <span
        data-testid="info"
        className="pagination-info"
      >{`${currentPage} of ${totalPage}`}</span>
      <button
        type="button"
        data-testid={NEXT_BUTTON_TEST_ID}
        className="btn"
        onClick={() => paginate(currentPage + 1)}
        disabled={currentPage === totalPage}
      >
        &gt;
      </button>
    </div>
  );
};

export default Pagination;
