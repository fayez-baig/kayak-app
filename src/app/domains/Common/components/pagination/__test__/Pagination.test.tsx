import { createRenderer } from "react-test-renderer/shallow";
import { render } from "@testing-library/react";
import fireEvent from "@testing-library/user-event";
import Pagination from "../Pagination";
import { NEXT_BUTTON_TEST_ID, PREV_BUTTON_TEST_ID } from "utils/constants";

const renderer = createRenderer();

const props = {
  itemsPerPage: 12,
  totalAirlinesData: 1306,
  paginate: jest.fn(),
  currentPage: 1,
};

describe("<Pagination />", () => {
  it("should render and match the snapshot", () => {
    renderer.render(<Pagination {...props} />);
    const renderedOutput = renderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });

  it("on click next button", () => {
    const { getByTestId } = render(<Pagination {...props} />);
    const btn = getByTestId(NEXT_BUTTON_TEST_ID);
    fireEvent.click(btn);
    expect(props.paginate).toHaveBeenCalled();
  });

  it("on click prev button", () => {
    props.currentPage = props.currentPage + 1;
    const { getByTestId } = render(<Pagination {...props} />);
    const btn = getByTestId(PREV_BUTTON_TEST_ID);
    fireEvent.click(btn);
    expect(props.paginate).toHaveBeenCalled();
  });
});
