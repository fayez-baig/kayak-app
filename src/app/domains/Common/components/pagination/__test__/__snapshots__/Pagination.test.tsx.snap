// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pagination /> should render and match the snapshot 1`] = `
<div
  className="pagination-container"
>
  <button
    className="btn"
    data-testid="prev-btn"
    disabled={true}
    onClick={[Function]}
    type="button"
  >
    &lt;
  </button>
  <span
    className="pagination-info"
    data-testid="info"
  >
    1 of 108
  </span>
  <button
    className="btn"
    data-testid="next-btn"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    &gt;
  </button>
</div>
`;
